drivers/phoenixcontact_modbus.c

```diff
@@ -77,49 +119,163 @@ void upsdrv_updateinfo(void)
 	}
 
 	if (tab_reg[1]) {
-		status_set("LB");	/* LB is actually called "shutdown event" on this ups */
+		status_set("LB");	
 	}
 
-	mrir(modbus_ctx, 29745, 1, tab_reg);
+	switch (UPSModel)
+	{
+	case NEW_UPS:
+
+		mrir(modbus_ctx, 0x2006, 1, tab_reg);
+		break;
+	case OLD_UPS:
+
+		mrir(modbus_ctx, 29745, 1, tab_reg);
+		break;
+	default:
+		break;
+	}
+	
 	dstate_setinfo("output.voltage", "%d", (int) (tab_reg[0] / 1000));
 
-	mrir(modbus_ctx, 29749, 5, tab_reg);
+	switch (UPSModel)
+	{
+	case NEW_UPS:
+
+		mrir(modbus_ctx, 0x200F, 1, tab_reg);
+		break;
+	case OLD_UPS:
+
+		mrir(modbus_ctx, 29749, 5, tab_reg);
+		break;
+	default:
+		break;
+	}
+		
 	dstate_setinfo("battery.charge", "%d", tab_reg[0]);
 	/* dstate_setinfo("battery.runtime",tab_reg[1]*60); */ /* also reported on this address, but less accurately */
 
-	mrir(modbus_ctx, 29792, 10, tab_reg);
+	uint16_t battery_voltage;
+	uint16_t battery_temperature;
+	uint16_t battery_runtime;
+	uint16_t battery_capacity;
+	uint16_t output_current;
+
+	switch (UPSModel)
+	{
+	case NEW_UPS:
+				
+		mrir(modbus_ctx, 0x200A, 1, &battery_voltage);
+		tab_reg[0] = battery_voltage;
+				
+		mrir(modbus_ctx, 0x200D, 1, &battery_temperature);
+		tab_reg[1] = battery_temperature;
+				
+		mrir(modbus_ctx, 0x2010, 1, &battery_runtime);
+		tab_reg[3] = battery_runtime;
+
+		mrir(modbus_ctx, 0x4211, 1, &battery_capacity);
+		tab_reg[8] = battery_capacity;
+
+		mrir(modbus_ctx, 0x2007, 1, &output_current);
+		tab_reg[6] = output_current;
+
+		break;
+	case OLD_UPS:
+
+		mrir(modbus_ctx, 29792, 10, tab_reg);
+		break;
+	default:
+		break;
+	}
+	
 	dstate_setinfo("battery.voltage", "%f", (double) (tab_reg[0]) / 1000.0);
 	dstate_setinfo("battery.temperature", "%d", tab_reg[1] - 273);
 	dstate_setinfo("battery.runtime", "%d", tab_reg[3]);
 	dstate_setinfo("battery.capacity", "%d", tab_reg[8] * 10);
 	dstate_setinfo("output.current", "%f", (double) (tab_reg[6]) / 1000.0);
 
 	/* ALARMS */
-	mrir(modbus_ctx, 29840, 1, tab_reg);
+	
 	alarm_init();
-	if (CHECK_BIT(tab_reg[0], 4) && CHECK_BIT(tab_reg[0], 5))
-		alarm_set("End of life (Resistance)");
-	if (CHECK_BIT(tab_reg[0], 6))
-		alarm_set("End of life (Time)");
-	if (CHECK_BIT(tab_reg[0], 7))
-		alarm_set("End of life (Voltage)");
-	if (CHECK_BIT(tab_reg[0], 9))
-		alarm_set("No Battery");
-	if (CHECK_BIT(tab_reg[0], 10))
-		alarm_set("Inconsistent technology");
-	if (CHECK_BIT(tab_reg[0], 11))
-		alarm_set("Overload Cutoff");
-	/* We don't use those low-battery indicators below.
-	 * No info or configuration exists for those alarm low-bat
-	 */
-	if (CHECK_BIT(tab_reg[0], 12))
-		alarm_set("Low Battery (Voltage)");
-	if (CHECK_BIT(tab_reg[0], 13))
-		alarm_set("Low Battery (Charge)");
-	if (CHECK_BIT(tab_reg[0], 14))
-		alarm_set("Low Battery (Time)");
-	if (CHECK_BIT(tab_reg[0], 16))
-		alarm_set("Low Battery (Service)");
+
+	uint32_t alarms_word;
```

Comment:
Here and above, variable definitions are added in the middle of scope. NUT coding style requires all definitions to be before code, so older systems can build current NUT too. Probably that is what CI failed about all over the place.

Reply:
Undersoot!